sql/src/main/java/org/apache/druid/sql/avatica/DruidStatement.java

```diff
@@ -306,11 +306,11 @@ public boolean isDone()
   @Override
   public void close()
   {
-    synchronized (lock) {
-      final State oldState = state;
-      state = State.DONE;
+    final State oldState = state;
```

Comment:
The `State oldState = state` assignment should happen inside `synchronized (lock)`, since otherwise it's an unsafe read.

Are the changes to `close()` here needed, given that the synchronization on DruidConnection.statements was removed from onClose?

Reply:
You're right this is an unsafe read, should be put into synchronized block